cypress/e2e/alerts.cy.js

```diff
@@ -1,25 +1,57 @@
 describe('Cypress application', () => {
-  before(() => {
-
+  beforeEach(() => {
+    cy.visit('https://demoqa.com/alerts')
   });
 
   it('should have the ability to assert automatically resolved alerts', () => {
-
+    cy.get('#alertButton')
+      .click();
+    cy.on('window:alert', (str) => {
+    expect(str).to.equal(`You clicked a button`)
+    })
   });
 
   it('should have the ability to assert scheduled allert', () => {
-
+    cy.get('#timerAlertButton')
+      .click();
+    cy.wait(5000);
+    cy.on('window:alert', (str) => {
+    expect(str).to.equal(`This alert appeared after 5 seconds`)
+    })
   });
 
-  it('should autimatically resolve alerts', () => {
-
-  });
+  it('should automatically resolve alerts', () => {
+    cy.get('#confirmButton')
+      .click();
+    cy.on('window:confirm', (str) => {
+      expect(str).to.equal(`Do you confirm action?`)
+      return true;
+    });
+    cy.get('#confirmResult')
+      .should('contain', 'You selected Ok')
+    });
 
   it('should have the ability to Cancel alerts', () => {
-
-  });
-
+    cy.get('#confirmButton')
+      .click();
+    cy.on('window:confirm', (str) => {
+      expect(str).to.equal(`Do you confirm action?`)
+      return false;
+    });
+    cy.get('#confirmResult')
+      .should('contain', 'You selected Cancel');
+    });
+  
   it('should have the ability to enter text to alert', () => {
-
+    cy.get('#promtButton')
+      .click();
+    cy.window().then((win) => {
```

Comment:
The correct actions should be at first to check if window exist and after that click on the button and enter text cy.window()
      .then((win) => {  
cy.get('#promtButton')
        .click()